tests/jballerina-unit-test/src/test/resources/test-src/javainterop/dependently_typed_functions_test.bal

```diff
@@ -51,6 +54,17 @@ function testSimpleTypes() {
     assert(true, b);
 }
 
+function testReferredTypes() {
```

Comment:
Without annotations, how does this test https://github.com/ballerina-platform/ballerina-lang/issues/36543#issuecomment-1156239421?

Reply:
Please add tests 
- with and without annotations and 
- when the referred type is another reference type and not a reference type

Reply:
> Without annotations, how does this test [#36543 (comment)](https://github.com/ballerina-platform/ballerina-lang/issues/36543#issuecomment-1156239421)?

The issue is regarding the type informations, not regarding annotations. With or Without annotations the type `Bar` will pass as `Bar` with this PR changes. Early, `Bar` type will pass as `int|float`. So type informations will be missed.

Reply:
> ## Purpose
> > Pass annotations for union type refernce types.
> 
> Fixes #36543

The issue you are fixing is https://github.com/ballerina-platform/ballerina-lang/issues/36543, right? 

The root cause may be how the type is handled and we need tests for that too. But if we are closing https://github.com/ballerina-platform/ballerina-lang/issues/36543 with this PR we need to make sure whatever the issue in it is also fixed.

Reply:
Can you check with L53 in annotation.bal test file and if there anything need to add as tests.

Reply:
We need to have a test similar to https://github.com/ballerina-platform/ballerina-lang/issues/36543#issuecomment-1156239421. The annotations on the inferred typedesc argument have to be asserted.

Reply:
The test you have added in annotation.bal works for me on update 3 RC1 also.

Reply:
Update tests

Reply:
Can you please point to where the annotation attachments are asserted?

Reply:
Annotation attachments are tested in `annotation.bal`.
In here tested the function argument type.

Reply:
> We need to have a test similar to [#36543 (comment)](https://github.com/ballerina-platform/ballerina-lang/issues/36543#issuecomment-1156239421). The annotations on the inferred typedesc argument have to be asserted.

This is still not tested?

Reply:
Added. Please check